src/Data/LSCA.hs

```diff
@@ -0,0 +1,112 @@
+-- | This module implements the Lowest Single Ancestor Tree (LSAT), as per
+-- Fischer 2010
+-- (https://www.sciencedirect.com/science/article/abs/pii/S0020019010000487).
+--
+-- All of the examples below are given over the following graph:
+--
+-- >   0
+-- >  / \
+-- > 1   2
+-- >  \ / \
+-- >   3   4
+--
+-- which is preprocessed into an 'LSAT' via:
+--
+-- $setup
```

Comment:
To clarify, a `$setup` block needs to be a standalone comment with `$setup` as the first line, or else `cabal-docspec` won't recognize it. (See the diff in https://github.com/ekmett/lca/pull/8#issuecomment-2615883168 for an example of what `cabal-docspec` is searching for.)

Reply:
Thanks for your patience with me! I don't have a working setup on my end and am trying to avoid doing so. The challenge afaict is that we'd also like this setup code to be part of the visible haddock.

Reply:
Unless you're suggesting that we copy and paste the initialization in both a $setup and the actual haddock? Which seems reasonable but kinda defeats the idea of a doctest imo :)

Reply:
> The challenge afaict is that we'd also like this setup code to be part of the visible haddock.

I don't think `cabal-docspec` allows this. (Unless I have overlooked something, @phadej?)

> Unless you're suggesting that we copy and paste the initialization in both a $setup and the actual haddock?

That might be the most expedient way to make this work.